src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -646,25 +631,28 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
 
     const isByronPath = (path: number[]) => classifyPath(path) === PathTypes.PATH_WALLET_SPENDING_KEY_BYRON
 
-    const byronWitnesses = ledgerWitnesses
-      .filter((witness) => isByronPath(witness.path))
-      .map((witness) => {
-        const { cborXPubKeyHex } = getSigningFileDataByPath(witness.path)
-        const { pubKey, chainCode } = splitXPubKeyCborHex(cborXPubKeyHex)
-        return TxByronWitness(pubKey, witness.signature, chainCode, {})
-      })
-
-    // TODO should be properly checked is the witness path is valid?
-    // TODO pool cold key witnesses?
-    const shelleyWitnesses = ledgerWitnesses
-      .filter((witness) => !isByronPath(witness.path))
-      .map((witness) => {
-        const { cborXPubKeyHex } = getSigningFileDataByPath(witness.path)
-        const { pubKey } = splitXPubKeyCborHex(cborXPubKeyHex)
-        return TxShelleyWitness(pubKey, witness.signature)
-      })
+    const [byronWitnesses, shelleyWitnesses] = partition(
```

Comment:
I think this deserves a comment on what is meant by "shelleyWitnesses". E.g. whether it includes pool cold key witnesses etc.